### ISSUE TITLE
Extends with no additional JSON properties

### ISSUE THREAD

**Issue by ArnaudGallardo** [GillianPerard/typescript-json-serializer#58]
Hi guys,
I have to serialize a class that extends another class but does not have any additional JSON properties, and that seems to break the lib. Here is a small reproducer:

```typescript
import { Serializable, JsonProperty, serialize } from 'typescript-json-serializer';

@Serializable()
export class Animal {
    @JsonProperty()
    public id: number;
    @JsonProperty()
    public name: string;
    public constructor(name: string) {
        this.name = name;
    }
}

@Serializable('Animal')
export class Dog extends Animal {
  public constructor(@JsonProperty() public color: string, name: string) {
    super(name);
  }
}

@Serializable('Animal')
export class Cat extends Animal {
  public constructor(name: string) {
    super(name);
  }
}

const clifford = new Dog('red', 'Clifford');
console.log(serialize(clifford)); // -> {color: "red", name: "Clifford"}

const garfield = new Cat('Garfield');
console.log(serialize(garfield)); // -> {}
```

I should have `{ name: "Garfield" }`, right?
Thanks!

**Comment by GillianPerard** [comment]
Hi, you're totally right!

I'll fix it.

**Comment by GillianPerard** [comment]
I fixed it in the 1.4.5 version!

I let you update and tell me if it's good.

Thank you for using my lib and for creating this issue 😊.
